pip/req/req_uninstall.py

```diff
@@ -131,60 +131,78 @@ def compact(self, paths):
         necessary to contain all paths in the set. If /a/path/ and
         /a/path/to/a/file.txt are both in the set, leave only the
         shorter path."""
+
+        sep = os.path.sep
         short_paths = set()
         for path in sorted(paths, key=len):
-            if not any([
-                    (path.startswith(shortpath) and
-                     path[len(shortpath.rstrip(os.path.sep))] == os.path.sep)
-                    for shortpath in short_paths]):
+            should_add = any(
+                path.startswith(shortpath.rstrip("*")) and
+                path[len(shortpath.rstrip("*").rstrip(sep))] == sep
+                for shortpath in short_paths
+            )
+            if not should_add:
```

Comment:
That's what I did initially but these same paths are being used to delete files and I didn't want to modify this logic because of that.